### PR TITLE
15814 - Staff (BCOL Help Desk, SBC and BC Registries) helping Client to affiliate on their behalf

### PR DESCRIPTION
**Pull request by TMunro7**
Issue #15814
https://github.com/bcgov/entity/issues/15814

*Description of changes:*
Add backend function that returns true if a user is a member of SBC_Staff, and returns false otherwise. additionally added function is_authorized which checks if a user is authorized and if the user is a member of staff or sbc_staff, then returns true.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/services/affiliation.py [hunk 1]
```diff
@@ -198,6 +184,26 @@ def create_affiliation(org_id, business_identifier, pass_code=None, certified_by
                                                            name=name, id=entity.business_identifier))
         return Affiliation(affiliation)
 
+    @staticmethod
+    @user_context
+    def is_authorized(entity: Entity, pass_code: str, **kwargs) -> bool:
+        """Return True if user is authorized to create an affiliation."""
+        user_context: UserContext = kwargs['user_context']
+        current_user: UserService = UserService.find_by_jwt_token(silent_mode=True)
```
Comment (seeker25) [inline auth-api/src/auth_api/services/affiliation.py:192]: Many unit tests failed without silent_mode.

**Comment by seeker25** [comment]
We borrowed quite a bit of the highlighting code that already existed in the feature branch. Special thanks to Entities for that.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2412-gp5ywkoa.web.app

File: auth-web/src/components/auth/manage-business/AddBusinessDialog.vue [hunk 2]
```diff
@@ -77,7 +77,7 @@
             <!-- Passcode -->
             <v-expand-transition>
               <v-text-field
-                v-if="isBusinessIdentifierValid"
+                v-if="isBusinessIdentifierValid && !isGovStaffAccount"
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/AddBusinessDialog.vue:80]: There's probably a way to wrap all of these with one v-if

File: auth-web/src/components/auth/manage-business/EntityManagement.vue [hunk 3]
```diff
@@ -362,18 +372,24 @@ export default class EntityManagement extends Mixins(AccountMixin, AccountChange
   private secondaryBtnHandler: () => void = undefined
   private lastSyncBusinesses = 0
   protected addBusinessDialog = false
+  snackbarText: string = null
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/EntityManagement.vue:375]: tried to make this similar to what is already in the feature branch

File: auth-api/tests/utilities/factory_utils.py [hunk 4]
```diff
@@ -433,3 +436,14 @@ def raise_for_status(self):
             pass
 
     monkeypatch.setattr('auth_api.services.rest_service.RestService.get', lambda *args, **kwargs: MockPartiesResponse())
+
+
+def convert_org_to_staff_org(org_id: int, type_code: OrgType):
```
Comment (ochiu) [inline auth-api/tests/utilities/factory_utils.py:441]: nice

File: auth-api/src/auth_api/services/affiliation_invitation.py [hunk 5]
```diff
@@ -135,8 +135,9 @@ def create_affiliation_invitation(affiliation_invitation_info: Dict,
 
         if affiliation_invitation.type != AffiliationInvitationType.PASSCODE.value:
             # Fetch the up-to-date business details from legal API
-            business = AffiliationInvitation._get_business_details(business_identifier,
-                                                                   RestService.get_service_account_token())
+            token = RestService.get_service_account_token(config_id='ENTITY_SVC_CLIENT_ID',
```
Comment (ochiu) [inline auth-api/src/auth_api/services/affiliation_invitation.py:138]: thanks for rolling this fix in

**Review by ochiu — APPROVED** [review]

**Comment by Jxio** [comment]
auth-web package version need to update

File: auth-web/src/components/auth/manage-business/AddBusinessDialog.vue [hunk 6]
```diff
@@ -172,238 +174,292 @@
...
+      // business id is required
+      // passcode is required
+      // firms must accept certify clause
+      // staff users must enter names
+      // validate the form itself (according to the components' rules/state)
+      return (
+        !!businessIdentifier.value &&
+        !!passcode.value &&
+        (!isFirm.value || isCertified.value) &&
+        (!!certifiedBy.value || !isFirm.value) &&
```
Comment (Jxio) [inline auth-web/src/components/auth/manage-business/AddBusinessDialog.vue:342]: (!isFirm.value || isCertified.value) && (!!certifiedBy.value || !isFirm.value) seem to have some redundancy. If isFirm.value is false, both conditions will be true, regardless of the other values.

Comment (seeker25) [reply auth-web/src/components/auth/manage-business/AddBusinessDialog.vue:342]: Should have been addressed in https://github.com/bcgov/sbc-auth/pull/2403

Comment (seeker25) [reply auth-web/src/components/auth/manage-business/AddBusinessDialog.vue]: `        (!isFirm.value || (isCertified.value && !!certifiedBy.value)) &&`

**Review by Jxio — APPROVED** [review]
looks good to me!

**Pull request merged**